lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1243,10 +1284,11 @@ struct FIRStmtParser : public FIRParser {
                          FIRModuleContext &moduleContext,
                          Namespace &modNameSpace, const FIRVersion &version)
       : FIRParser(moduleContext.getConstants(), moduleContext.getLexer()),
-        builder(mlir::ImplicitLocOpBuilder::atBlockEnd(
-            UnknownLoc::get(getContext()), &blockToInsertInto)),
+        builder(UnknownLoc::get(getContext()), getContext()),
         locationProcessor(this->builder), moduleContext(moduleContext),
-        modNameSpace(modNameSpace), version(version) {}
+        modNameSpace(modNameSpace), version(version) {
+    builder.setInsertionPointToEnd(&blockToInsertInto);
```

Comment:
What's going on here?

Reply:
Below, I:
1. create a block
2. parse statements into it
3. once all blocks are parsed, create the match op and insert the blocks.

When you create a builder from `atBlockEnd(Block*block)`, the builder extracts a `MLIRContext*` by reaching into the op that owns the block (`block->getParent()->getContext()`).  The problem is that my block does not have an owner yet.  So I am fixing this by creating the builder with a known context, and then separately setting the insertion point.

There is a small ordering problem where I have to know the number of regions before I can create the operation.  Since I don't want to scan far ahead to know the regions, I parse the statements in to an unowned block.

Reply:
Okay, ty for the great explanation!